### PR TITLE
Add Sage Rails Switch and align to spec

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description
- [x] - aligned Switch component to spec
- [x] - add Sage Rails Switch component

### Screenshots
|  before  |  after  |
|--------|--------|
|Screen Shot 2020-10-20 at 3 50 13 PM|Screen Shot 2020-10-20 at 3 49 38 PM|

### Steps for testing
1. Visit the Sage Switch page: http://localhost:4000/pages/element/switch

### CODE REVIEW

File: lib/sage-frontend/stylesheets/system/core/_variables.scss [hunk 1]
```diff
@@ -217,7 +217,7 @@ $sage-switch-toggle-size: rem(20px) !default;
 
 // Focus state
```
Comment (pixelflips) [inline lib/sage-frontend/stylesheets/system/core/_variables.scss]: Could these focus state styles also possibly be moved into the _switch.scss file?

File: lib/sage_rails/app/sage_components/sage_switch.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  attr_accessor :type
+  attr_accessor :id
+  attr_accessor :name
+  attr_accessor :label_text
+  attr_accessor :value
+  attr_accessor :checked
+  attr_accessor :disabled
+  attr_accessor :required
+  attr_accessor :message
+end
```
Comment (pixelflips) [inline lib/sage_rails/app/sage_components/sage_switch.rb]: Missing newline here.

File: lib/sage_rails/app/views/sage_components/_sage_switch.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    value="<%= component.id %>"
+    <%= component.checked  ? " checked" : "" %>
+    <%= component.disabled ? " disabled" : "" %>
+    <%= component.required ? " required" : "" %>
+  >
+  <label for="<%= component.id %>" class="sage-switch__label"><%= component.label_text %></label>
+  <% if component.message != "" %>
+    <div class="sage-switch__message">Additional Info</div>
+  <% end %>
+</div>
```
Comment (pixelflips) [inline lib/sage_rails/app/views/sage_components/_sage_switch.html.erb]: Missing newline here.

**Review by pixelflips — APPROVED** [review]
👍  A couple of comments, but nothing show-stopping. Great work!

**Review by teenwolfblitzer — APPROVED** [review]
👍  LGTM, nice work!

**Review by AndrwM — APPROVED** [review]
This is looking really great 😌 +1

**Pull request merged**